lib/integrations/awsoidc/idp_iam_config.go

```diff
@@ -154,56 +328,267 @@ func ConfigureIdPIAM(ctx context.Context, clt IdPIAMConfigureClient, req IdPIAMC
 		req.AccountID = aws.ToString(callerIdentity.Account)
 	}
 
-	thumbprint, err := ThumbprintIdP(ctx, req.ProxyPublicAddress)
-	if err != nil {
+	logrus.Infof("Creating IAM OpenID Connect Provider: url=%q.", req.issuerURL)
+	if err := ensureOIDCIdPIAM(ctx, clt, req); err != nil {
+		return trace.Wrap(err)
+	}
+
+	logrus.Infof("Creating IAM Role %q.", req.IntegrationRole)
+	if err := upsertIdPIAMRole(ctx, clt, req); err != nil {
 		return trace.Wrap(err)
 	}
-	logrus.Infof("Using the following thumbprint: %s", thumbprint)
 
-	createOIDCResp, err := clt.CreateOpenIDConnectProvider(ctx, &iam.CreateOpenIDConnectProviderInput{
+	// Configuration stops here if there's no S3 bucket.
+	// It will use the teleport's public address as IdP issuer.
+	if req.s3Bucket == "" {
+		return nil
+	}
+	log := logrus.WithFields(logrus.Fields{
+		"bucket":        req.s3Bucket,
+		"bucket-prefix": req.s3BucketPrefix,
+	})
+
+	log.Infof("Creating bucket in region %q", clt.CreateBucketRegion())
```

Comment:
`ensureBucketIdPIAM()` on the next line can change the bucket region though?

Reply:
If we get an `AlreadyExists` error, we print the bucket's region and continue.
I think that's enough to indicate that the intention was to create in region X, but we ended up using region Y because it already exists.

Do you think we should have a more clear message? Or removing the region in that log line?